bokeh/core/property/container.py

```diff
@@ -361,17 +361,22 @@ class RestrictedDict(Dict):
 
     """
 
+    def __init__(self, disallow, keys_type, values_type, default={}, help=None):
+        self._disallow = disallow
+        super.__init__(self, keys_type, values_type, default={}, help=None)
+        
     def validate(self, value, detail=True):
         super().validate(value, detail)
 
-        disallowed_keys = self.disallow
+        disallowed_keys = self._disallow
         user_keys = self.keys()
-
-        for k in user_keys:
-            if k in disallowed_keys:
-                msg = f"{k} key is disallowed"
-                raise ValueError(msg)
-            break
+        
+        error_keys = list(set(disallowed_keys) & set(user_keys))
```

Comment:
suggest creating the disallowed set once in the init:
```
self._disallow = set(disallow)
```
then this can be:
```
error_keys = self._disallow & value.keys()
```

The the temporary assignments at the top can also be removed. 

Note also `value.keys()` not `self.keys()` The value of `self` is the property type, not an actual dict value. The dict value is passed in as `value` to the `validate` method to check.